packages/preview/modernpro-coverletter/0.0.1/modernpro-coverletter.typ

```diff
@@ -0,0 +1,204 @@
+    }
+  }
+
+  let font_info = {
+    if fonttype == "macfont" {
+      "Helvetica"
+    } else if fonttype == "openfont" {
+      "PT Sans"
+    } else {
+      "Times New Roman"
+    }
+  }
+
+  let recipientgenerate(starttitle, cltitle, date, department, institution, address, postcode) = {
+    align(
+      left,
+      {
+        if department != [] {
+          text(
+            10pt,
+            font: font_info,
+            fill: subheadings_colour,
+            weight: "bold",
+          )[#department]
+        }
+        h(1fr)
+        if date != "" {
+          text(10pt, font: font_info, fill: primary_colour, weight: "light")[#date\ ]
+        } else {
+          text(
+            10pt,
+            font: font_info,
+            fill: primary_colour,
+            weight: "light",
+          )[ #datetime.today(offset: auto).display("[day] [month repr:long] [year]")\ ]
+        }
+
+        if institution != [] {
+          text(
+            10pt,
+            font: font_info,
+            fill: subheadings_colour,
+            weight: "bold",
+          )[#institution\ ]
+        }
+
+        if address != [] {
+          text(
+            10pt,
+            font: font_info,
+            fill: headings_colour,
+            weight: "light",
+          )[#address\ ]
+        }
+        if postcode != [] {
+          text(
+            10pt,
+            font: font_info,
+            fill: headings_colour,
+            weight: "light",
+          )[#postcode ]
+        }
+      },
+    )
+
+
+    align(left, text(
+      12pt,
+      font: font_head,
+      fill: primary_colour,
+      weight: "medium",
+    )[#upper(cltitle)])
+    v(0.1em)
+    set text(11pt, font: font_head, fill: primary_colour, weight: "regular")
+
+    [#starttitle]
+  }
+
+  // show contact details
+  let display(contacts) = {
+    set text(
+      11pt,
+      font: font_term,
+      fill: headings_colour,
+      weight: "medium",
+      top-edge: "baseline",
+      bottom-edge: "baseline",
+      baseline: 2pt,
+    )
+    contacts.map(contact =>{
+      if contact.link == none [
+        contact.text
+      ] else {
+        link(contact.link)[#{ contact.text }]
+      }
+    }).join(" | ")
+  }
+
+  set page(margin: (left: 2cm, right: 2cm, top: 3.2cm, bottom: 1.5cm), header: {
+    // Head Name Section
+    text(
+      25pt,
+      font: font_head,
+      fill: primary_colour,
+      weight: "light",
+      top-edge: "baseline",
+      bottom-edge: "baseline",
+      baseline: 12pt,
+    )[#align(center, [#name])]
+    text(
+      11pt,
+      font: font_descript,
+      fill: headings_colour,
+      weight: "medium",
+      top-edge: "baseline",
+      bottom-edge: "baseline",
+    )[#align(center, [#address])]
+    align(center)[#display(contacts)]
+    line(length: 100%, stroke: 0.5pt + primary_colour)
+  }, header-ascent: 1em)
+
+  // Add recipient details
+  recipientgenerate(
+    recipient.starttitle,
+    recipient.cltitle,
```

Comment:
I get this error upon initialization:
```
error: dictionary does not contain key "cltitle"
    ┌─ @preview/modernpro-coverletter:0.0.1/modernpro-coverletter.typ:186:14
    │
186 │     recipient.cltitle,
    │               ^^^^^^^
```

Reply:
Thanks! I fixed the error and tested the template from my side.